data/random-teams.ts

```diff
@@ -1731,18 +1730,37 @@ export class RandomTeams {
 					if (!species.types.includes(type)) continue;
 				}
 			}
-			pokemonPool.push(pokemon);
-			baseSpeciesCount[species.baseSpecies] = (baseSpeciesCount[species.baseSpecies] || 0) + 1;
+
+			if (species.baseSpecies in pokemonPool) {
+				pokemonPool[species.baseSpecies].count++;
+				pokemonPool[species.baseSpecies].speciesArray.push(species.id);
+			} else {
+				pokemonPool[species.baseSpecies] = {count: 1, speciesArray: [species.id]};
+			}
```

Comment:
At this point `count` is always the same as `speciesArray.length`...
```suggestion
			if (species.baseSpecies in pokemonPool) {
				pokemonPool[species.baseSpecies].push(species.id);
			} else {
				pokemonPool[species.baseSpecies] = [species.id];
			}
```